src/libstd/primitive_docs.rs

```diff
@@ -10,8 +10,46 @@
 
 #[doc(primitive = "bool")]
 //
-/// The boolean type.
+/// The boolean type represents a value, which could only be either `true` or `false`.
 ///
+/// It implements the various traits, such as [`BitAnd`], [`BitOr`], [`Not`], etc., to deal with
+/// the boolean operations. [`if`] always demands a `bool` value.
```

Comment:
And I think the last half of this sentence can just go away. Implementing the traits are enough.

Furthermore, the comment about `if` feels like it should be on its own line, as it's something distinct.